cmake/thirdparty/get_spdlog.cmake

```diff
@@ -17,11 +17,16 @@ function(find_and_configure_spdlog)
 
   include(${rapids-cmake-dir}/cpm/spdlog.cmake)
   rapids_cpm_spdlog(INSTALL_EXPORT_SET rmm-exports)
+  rapids_export_package(BUILD spdlog rmm-exports)
 
   if(spdlog_ADDED)
-    install(TARGETS spdlog_header_only EXPORT rmm-exports)
-  else()
-    rapids_export_package(BUILD spdlog rmm-exports)
+    rapids_export(
+      BUILD spdlog
+      EXPORT_SET spdlog
+      GLOBAL_TARGETS spdlog spdlog_header_only
+      NAMESPACE spdlog::)
+    include("${rapids-cmake-dir}/export/find_package_root.cmake")
+    rapids_export_find_package_root(BUILD spdlog [=[${CMAKE_CURRENT_LIST_DIR}]=] rmm-exports)
```

Comment:
For my edification, what does the `[=.*=]` syntax do? 

Do we want this path to be relative to the source/build/install root? It's going to be a header-only library that gets installed as part of rmm (after rmm pulls it down via CPM), right?

Reply:
I don't understand why you don't want to evaluate it. Does that mean that inside `rapids_export_find_package_root` instead of the value of `CMAKE_CURRENT_LIST_DIR` the variable name will be propagated through and then it will be evaluated only when another package actually tries to get the spdlog target from rmm?

Reply:
> the variable name will be propagated through and then it will be evaluated only when another package actually tries to get the spdlog target from rmm?

Correct. This code snippet is injected into the rmm-config.cmake and so we want it to evaluate when that file is parsed, and not when `rapids_export_find_package_root` is called.